security/auth/maskinporten.md

```diff
@@ -0,0 +1,141 @@
+
+### Access Policies
+
+The following [outbound external hosts](../../nais-application/access-policy.md#external-services) are automatically added when enabling this feature:
+
+- `ver2.maskinporten.no` in development
+- `maskinporten.no` in production
+
+You do not need to specify these explicitly.
+
+### Scopes
+
+Maskinporten allows API providers to define access to their APIs, modeled as scopes and based on the consumer's organization number.
+
+When a `client` requests Maskinporten (Identity Provider) for a token:
+- Maskinporten validates the validity of the JWT and signature ([Runtime JWK Secret](#runtime-variables-&-credentials) used to sign the JWT).  
+- When `client` has access to the requested resources: `scope`, an `access_token` will be returned to the `client` and can be used for further actions.
+
+For more details see: [Overview of flows in Digdirator](#overview-of-flows)
+
+{% hint style="warning" %}
+Make sure **NAV** have `pre-registered` rights to **all** `scopes` specified in the manifest or provision o client will fail.
+NAV´s `pre-registered` scopes can be found with proper access rights in [Digdir selvbetjening](https://selvbetjening-samarbeid-ver2.difi.no/auth/login).
+{% endhint %}
+
+### Runtime Variables & Credentials
+
+The following environment variables and files (under the directory `/var/run/secrets/nais.io/maskinporten`) are available at runtime:
+
+{% code-tabs %}
+{% code-tabs-item title="Description" %}
+| Name | Description |
+|---|---|
+| `MASKINPORTEN_CLIENT_ID` | Maskinporten client ID. Unique ID for the application in Maskinporten |
+| `MASKINPORTEN_CLIENT_JWK` | Private JWK containing the private RSA key for creating signed JWTs when using the [JWT grants](https://difi.github.io/felleslosninger/maskinporten_protocol_token.html). |
+| `MASKINPORTEN_SCOPES` |  The scopes registered for the client at Maskinporten as a whitepace-separated string. See [JWT grants](https://difi.github.io/felleslosninger/maskinporten_protocol_token.html) for more information. |
+| `MASKINPORTEN_WELL_KNOWN_URL` | The well-known URL for the OIDC metadata discovery document for Maskinporten. |
+{% endcode-tabs-item %}
+{% code-tabs-item title="Example values" %}
+| Name | Values |
+|---|---|
+| `MASKINPORTEN_CLIENT_ID` | `e89006c5-7193-4ca3-8e26-d0990d9d981f` |
+| `MASKINPORTEN_SCOPES` | `nav:first/scope nav:another/scope` |
+| `MASKINPORTEN_WELL_KNOWN_URL` | `https://ver2.maskinporten.no/.well-known/oauth-authorization-server` |
+{% endcode-tabs-item %}
+{% code-tabs-item title="Example value for MASKINPORTEN_CLIENT_JWK" %}
+```json
+{
+  "use": "sig",
+  "kty": "RSA",
+  "kid": "jXDxKRE6a4jogcc4HgkDq3uVgQ0",
+  "alg": "RS256",
+  "n": "xQ3chFsz...",
+  "e": "AQAB",
+  "d": "C0BVXQFQ...",
+  "p": "9TGEF_Vk...",
+  "q": "zb0yTkgqO...",
+  "dp": "7YcKcCtJ...",
+  "dq": "sXxLHp9A...",
+  "qi": "QCW5VQjO..."
+}
+```
+{% endcode-tabs-item %}
+{% endcode-tabs %}
+
+### Migration guide - step by step
```

Comment:
Migrating from where? Should clarify that this only applies to clients that already exist in the IaC repository.